pkg/server/image_pull.go

```diff
@@ -265,34 +264,43 @@ func (c *criContainerdService) pullImage(ctx context.Context, ref string) (
 	// 2) We need desc returned by schema1 converter.
 	// So just put the image metadata after downloading now.
 	// TODO(random-liu): Fix the potential garbage collection race.
-	if err = c.imageStoreService.Put(ctx, ref, desc); err != nil {
-		return "", "", fmt.Errorf("failed to put image %q desc %v into containerd image store: %v",
-			ref, desc, err)
+	repoDigest, repoTag := getRepoDigestAndTag(namedRef, desc.Digest)
+	if ref != repoTag && ref != repoDigest {
+		return "", "", "", fmt.Errorf("unexpected repo tag %q and repo digest %q for %q", repoTag, repoDigest, ref)
```

Comment:
Because in code below, we use `ref` to get image metadata from the containerd image store, which is making the assumption that `ref` is either `repoTag` or `repoDigest`.

Initially, I thought I should add a comment here, but then I remember one time I read in the toilet that sanity check is better than comment. So...